packages/arb-rpc-node/batcher/sequencerBatcher.go

```diff
@@ -508,10 +509,27 @@ func (b *SequencerBatcher) deliverDelayedMessages(chainTime inbox.ChainTime) (bo
 			return false, err
 		}
 	}
-	delayedAcc, err := b.db.GetDelayedInboxAcc(new(big.Int).Sub(newDelayedCount, big.NewInt(1)))
+	lastDelayedSeqNum := new(big.Int).Sub(newDelayedCount, big.NewInt(1))
+	delayedAcc, err := b.db.GetDelayedInboxAcc(lastDelayedSeqNum)
 	if err != nil {
 		return false, err
 	}
+	_, isSimulatedBackend := b.client.(*ethutils.SimulatedEthClient)
+	var getDelayedAccTarget *big.Int
+	if isSimulatedBackend {
+		// The simulated backend doesn't support querying against old blocks
+		getDelayedAccTarget = chainTime.BlockNum.AsInt()
+	} else {
+		// Confirm that the message wasn't reorganized forwards to a further reorganizable block
+		getDelayedAccTarget = lastConfirmedL1Block
+	}
+	l1DelayedAcc, err := b.inboxReader.GetDelayedAccumulator(ctx, lastDelayedSeqNum, getDelayedAccTarget)
```

Comment:
We should discuss whether adding an archive node requirement here is the only way to check this. In particular, could we use an event query instead?

Reply:
I believed that geth would keep the last 255 blocks in memory or so. This only queries from 12 blocks ago. Is that an issue? I can update it to use an event query if so.

Reply:
Ah good point, I hadn't considered that the backwards looking would be tightly bounded. We'll definitely want to verify that our geth node, alchemy, and infura can handle this, but it seems safe

Reply:
Alchemy, infura, and default geth all look good